cql3/statements/select_statement.cc

```diff
@@ -2032,7 +2032,10 @@ std::unique_ptr<prepared_statement> select_statement::prepare(data_dictionary::d
             && !restrictions->need_filtering()  // No filtering
             && group_by_cell_indices->empty()   // No GROUP BY
             && db.get_config().enable_parallelized_aggregation()
-            && !is_local_table();
+            && !is_local_table()
+            && !( // Do not parallelize the request if it's single partition read
+                restrictions->partition_key_restrictions_is_all_eq() 
+                && restrictions->partition_key_restrictions_size() == schema->partition_key_size());
```

Comment:
I really hope that if I type "p1 == 3 AND p1 == 4" (note the same p1 here), this doesn't get counted as size==2 and reach this case :-(
Again, only a test can show me it isn't a problem.

Reply:
`partition_key_restrictions_size()'` implementation seem to be using a `std::set` to deduplicate column definitions gathered when traversing the expression, so this shouldn't be a problem.

By the way, the method `partition_key_restrictions_size` itself is a mess - its name doesn't really align with its semantics it just returns the count of partition key columns used in the restrictions, not the number (size???) of partition key restrictions. Ideally, it should be given a clearer name and we should write tests for it - but maybe in a followup.

Reply:
It isn't parallelized. I'll add it to the test